### PR TITLE
feat(route53resolver): added postObserve status

### PR DESCRIPTION
**Pull request by haarchri**
Signed-off-by: haarchri <chhaar30@googlemail.com>

### Description of your changes

old status:
```
NAME                                                                  READY   SYNCED   EXTERNAL-NAME
resolverrule.route53resolver.aws.crossplane.io/sample-resolver-rule   False   True     rslvr-rr-d64a11650e394f618

NAME                                                                 READY   SYNCED   EXTERNAL-NAME
resolverendpoint.route53resolver.aws.crossplane.io/sample-resolver   False   True     rslvr-out-a8fd90a254954092b
```

new status:
```
NAME                                                                  READY   SYNCED   EXTERNAL-NAME
resolverrule.route53resolver.aws.crossplane.io/sample-resolver-rule   True    True     rslvr-rr-d64a11650e394f618

NAME                                                                 READY   SYNCED   EXTERNAL-NAME
resolverendpoint.route53resolver.aws.crossplane.io/sample-resolver   True    True     rslvr-out-a8fd90a254954092b
```

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

```
kubectl apply -f examples/route53resolver/resolverendpoint/resolverendpoint.yaml 
kubectl apply -f examples/route53resolver/resolverrule/resolvererule.yaml 
```

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: pkg/controller/route53resolver/resolverrule/hooks.go [hunk 1]
```diff
@@ -73,3 +76,20 @@ func preUpdate(_ context.Context, cr *v1alpha1.ResolverRule, obj *svcsdk.UpdateR
...
+
+func postObserve(_ context.Context, cr *svcapitypes.ResolverRule, obj *svcsdk.GetResolverRuleOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch aws.StringValue(obj.ResolverRule.Status) {
+	case string(svcapitypes.ResolverRuleStatus_SDK_COMPLETE):
+		cr.SetConditions(xpv1.Available())
+	case string(svcapitypes.ResolverRuleStatus_SDK_UPDATING), string(svcapitypes.ResolverRuleStatus_SDK_FAILED):
```
Comment (muvaf) [inline pkg/controller/route53resolver/resolverrule/hooks.go:88]: Are we sure the resource is not available during update?

File: pkg/controller/route53resolver/resolverendpoint/hooks.go [hunk 2]
```diff
@@ -87,3 +90,22 @@ func preUpdate(_ context.Context, cr *v1alpha1.ResolverEndpoint, obj *svcsdk.Upd
...
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch aws.StringValue(obj.ResolverEndpoint.Status) {
+	case string(svcapitypes.ResolverEndpointStatus_SDK_OPERATIONAL):
+		cr.SetConditions(xpv1.Available())
+	case string(svcapitypes.ResolverEndpointStatus_SDK_CREATING):
+		cr.SetConditions(xpv1.Creating())
+	case string(svcapitypes.ResolverEndpointStatus_SDK_UPDATING), string(svcapitypes.ResolverEndpointStatus_SDK_ACTION_NEEDED):
```
Comment (muvaf) [inline pkg/controller/route53resolver/resolverendpoint/hooks.go:104]: Are we sure the resource is not available during update or action needed?

**Review by muvaf — APPROVED** [review]
LGTM once the comments are addressed. Thanks @haarchri !

File: pkg/controller/route53resolver/resolverendpoint/hooks.go (hunk 2 above)
Comment (haarchri) [reply pkg/controller/route53resolver/resolverendpoint/hooks.go:104]: UPDATING: Resolver is associating or disassociating one or more network interfaces with this endpoint / yes If all addresses are updating 

ACTION_NEEDED: This endpoint is unhealthy, and Resolver can't automatically recover it. / yes

File: pkg/controller/route53resolver/resolverrule/hooks.go (hunk 1 above)
Comment (haarchri) [reply pkg/controller/route53resolver/resolverrule/hooks.go:88]: yes we have cases if we change eg. the targetIPs

**Pull request merged**